sql/information_schema/routines_table.go

```diff
@@ -234,7 +241,12 @@ func parametersRowIter(ctx *Context, c Catalog, p map[string][]*plan.Procedure)
 	if err != nil {
 		return nil, err
 	}
-	for db, procedures := range p {
+	privSet, _ := ctx.GetPrivilegeSet()
+	for dbName, procedures := range p {
+		if !(privSet.Has(PrivilegeType_CreateRoutine) || privSet.Has(PrivilegeType_AlterRoutine) || privSet.Has(PrivilegeType_Execute) ||
+			privSet.Database(dbName).Has(PrivilegeType_CreateRoutine) || privSet.Database(dbName).Has(PrivilegeType_AlterRoutine) || privSet.Database(dbName).Has(PrivilegeType_Execute)) {
```

Comment:
(minor) might be worth putting this into a helper function – it would help the code read a little smoother and if we need to update this in the future with new permission types/rules/checks then we'll have the code in one place and won't have to remember to update multiple places.